magento/magento2
# fix issue #24366 - Updating cart quantity

### Description (*)

This **Pull Request** fix Updating cart quantity, causes wrong error message to be displayed when backorders is involved. Internal ticket **MC-19752**

### Fixed Issues (if relevant)

1. magento/magento2#24366: Updating cart with quantity that results in error, causes wrong error message to be displayed when backorders involved.

### Manual testing scenarios (*)


1. Deploy **2.3-develop**
2. Create a test **Category**
3. Create test **product** in test **category.**
4. Set the stock **product** to Qty = 10
5. **Allows Backorders** with _Notify_ to the product
6. Set **Max in cart** = 30 in test product.
7. Go to frontend and add 10 items to cart
8. **Verify** no warning messages
9. Change quantity to 25 and Update cart
10. Verify backorder warning of 15 items backordered
11. Change quantity to 35 and Update cart

### Questions or comments


### Contribution checklist (*)
